test/ProductConstructionService.ScenarioTests/ScenarioTestBase.cs

```diff
@@ -1036,26 +1036,31 @@ protected string GetUniqueAssetName(string packageName)
     protected static IAsyncDisposable CleanUpPullRequestAfter(string owner, string repo, Octokit.PullRequest pullRequest)
         => AsyncDisposable.Create(async () =>
         {
-            try
-            {
-                var pullRequestUpdate = new Octokit.PullRequestUpdate
-                {
-                    State = Octokit.ItemState.Closed
-                };
+            await ClosePullRequest(owner, repo, pullRequest);
+        });
 
-                await GitHubApi.Repository.PullRequest.Update(owner, repo, pullRequest.Number, pullRequestUpdate);
-                await GitHubApi.Git.Reference.Delete(owner, repo, $"heads/{pullRequest.Head.Ref}");
-            }
-            catch
+    protected static async Task ClosePullRequest(string owner, string repo, Octokit.PullRequest pullRequest)
+    {
+        try
+        {
+            var pullRequestUpdate = new Octokit.PullRequestUpdate
             {
-                // Closed already
-            }
-        });
+                State = Octokit.ItemState.Closed
+            };
 
-    protected static async Task CreateTargetBranchAndExecuteTest(string targetBranchName, TemporaryDirectory targetDirectory, Func<Task> test)
+            await GitHubApi.Repository.PullRequest.Update(owner, repo, pullRequest.Number, pullRequestUpdate);
+            await GitHubApi.Git.Reference.Delete(owner, repo, $"heads/{pullRequest.Head.Ref}");
```

Comment:
If the first fails because the PR is closed already, we won't remove the branch. Maybe there should be 2 try/catch blocks?

Reply:
added